### DISCUSSION TITLE
Delay while editing, especially while cropping

### DISCUSSION THREAD

**Discussion by ocrunner84** [rsjaffe/MIDI2LR#670] [Problems]
Hi there,
am I the only one that notices some delay/latency when the knobs are used (Behringer X-Touch Mini), especially when I'm cropping?
I mean, I use two knobs to control cropping: one to the straighten angle and the other to crop in/out.
The issue is that is it a lot more snappy/responsive to use the mouse to crop than to use the midi controllers' knobs.
Maybe most people are ok with that (if they have or notice the issue) but in my case it slows me down a lot, because I edit many hundreds, sometimes thousands of pictures per day.

My computer isn't the fastest (Xeon x5670, 18GB RAM, nvme 970 Evo +...) but it is responsive enough if I use the mouse to crop.
Is this normal, because there's some "communication" between the MIDI controller, the plugin, and Lightroom, or is there anything that I can do to try to fix it?

Thanks in advance, the work done by Rory is amazing, I've done my donation as a way to say thanks for this amazing plugin.

**Comment by rsjaffe** [comment]
There’s a lot of work being done behind the scenes to make the crop knob work while preserving the preset aspect ratio, as Lightroom doesn’t implement that directly for plugins. I notice a little lag, and I currently have a fairly fast  machine. If Adobe updated their api to support proportional cropping, this issue would go away, but I doubt they ever will.

**Reply by ocrunner84** [reply]
I didn't know that.
So, basically, what you're saying is that every time that I use the knobs to crop, the plugin has to do some extra "work" to keep the aspect ratio?
Sorry but I know nothing about coding, but now it makes sense that the difference in responsiveness between cropping with the mouse vs using a MIDI controller is basically because of the extra steps that the computer has to do?
This is more of a CPU task, right? I've noticed while cropping that the CPU usage gets higher when I'm cropping fast.
I'll update my computer soon, maybe I'll see some improvements.
Do you think that devices like Loupedeck also have the same issues?

Thanks again for all the work and patience from you.
